pkg/grafana/sso_settings.go

```diff
@@ -14,57 +14,98 @@ const (
 	grafanaAdminRole  = "Admin"
 	grafanaEditorRole = "Editor"
 	grafanaViewerRole = "Viewer"
+
+	// ssoProvider is the OAuth provider type used for SSO configuration
+	ssoProvider = "generic_oauth"
 )
 
+// ConfigureSSOSettings configures Grafana SSO settings with organization mappings.
+// It retrieves the current SSO provider settings, updates the org_mapping field
+// with the provided organizations, and applies the changes to Grafana.
 func (s *Service) ConfigureSSOSettings(ctx context.Context, organizations []Organization) error {
-	logger := log.FromContext(ctx)
+	logger := log.FromContext(ctx).WithValues("provider", ssoProvider)
+
+	if len(organizations) == 0 {
+		logger.Info("no organizations provided, skipping SSO configuration")
+		return nil
+	}
 
-	provider := "generic_oauth"
-	resp, err := s.grafanaAPI.SsoSettings.GetProviderSettings(provider, nil)
+	resp, err := s.grafanaAPI.SsoSettings.GetProviderSettings(ssoProvider, nil)
 	if err != nil {
-		logger.Error(err, "failed to get sso provider settings.")
-		return errors.WithStack(err)
+		return errors.WithStack(fmt.Errorf("failed to get SSO provider settings for %s: %w", ssoProvider, err))
+	}
+
+	if resp.Payload == nil {
+		return errors.WithStack(fmt.Errorf("received nil payload from SSO provider settings for %s", ssoProvider))
+	}
+
+	// Safe type assertion with error handling
+	settings, ok := resp.Payload.Settings.(map[string]any)
+	if !ok {
+		return errors.WithStack(fmt.Errorf("unexpected settings type for %s: expected map[string]any, got %T", ssoProvider, resp.Payload.Settings))
+	}
+
+	orgsMapping, err := generateGrafanaOrgsMapping(organizations)
+	if err != nil {
+		return errors.WithStack(fmt.Errorf("failed to generate organization mappings for %s: %w", ssoProvider, err))
 	}
 
-	orgsMapping := generateGrafanaOrgsMapping(organizations)
-	settings := resp.Payload.Settings.(map[string]interface{})
-	settings["role_attribute_path"] = "to_string('Viewer')"
-	settings["org_attribute_path"] = "groups"
```

Comment:
These are not useful anymore?
This goes beyond refactoring and error handling.
Please make sure to write info about that in the PR description and the changelog, so it's easier to trace in case it breaks something.